chore: use test dependencies from node_modules

DanielRuf opened the pull request:
This replaces https://github.com/zurb/foundation-sites/pull/11177 and updates the browser-based test file to use the dependencies from node_modules.

Commit:
chore: update the browserslist config according to the compability table

DanielRuf committed:
chore: drop support for Android 2.3 - 4.4

chore: add note about old iOS versions

ncoden committed:
docs: improve README design & organization

Changes
* Add Foundation logo
* Add menu bar with installation/docs/releases/contributing
* Simplify local running & testing sections

Commit:
tests: prevent race condition in _reflow test

Commit:
chore: use test dependencies from node_modules

dependabot-support committed:
build: bump autoprefixer from 8.2.0 to 8.3.0

Bumps [autoprefixer](https://github.com/postcss/autoprefixer) from 8.2.0 to 8.3.0.
- [Release notes](https://github.com/postcss/autoprefixer/releases)
- [Changelog](https://github.com/postcss/autoprefixer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/postcss/autoprefixer/compare/8.2.0...8.3.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

ncoden commented on package.json:
```diff
@@ -39,6 +39,7 @@
     "babel-preset-env": "^1.6.1",
     "browser-sync": "^2.18.12",
     "browserstack-runner": "^0.7.0",
+    "chai": "^4.1.2",
```
We must update both `package-lock.json` and `yarn.lock`. Please do the following:
```bash
rm package-lock.json
yarn
npm i --package-lock-only
sed -i '' 's/http\:\/\//https\:\/\//g' package-lock.json
sed -i '' 's/http\:\/\//https\:\/\//g' yarn.lock
```

DanielRuf replied:
Hm, thought we use snyp or something similar. Will do it when I come to the installation of yarn.

ncoden replied:
I took care of that.

ncoden commented on test/javascript/index.html:
```diff
@@ -6,16 +6,16 @@
-    <script src="lib/chai.js"></script>
-    <script src="lib/chai-jquery.js"></script>
-    <script src="lib/sinon.js"></script>
+    <script src="../../node_modules/chai/chai.js"></script>
```
We use `chai` directly, so please install `chai` in devDependencies.

DanielRuf replied:
That's what I did (see the package.json)?

ncoden replied:
hmmm I did not seen that, sorry.

dependabot-support committed:
build: bump chalk from 2.3.2 to 2.4.0

Bumps [chalk](https://github.com/chalk/chalk) from 2.3.2 to 2.4.0.
- [Release notes](https://github.com/chalk/chalk/releases)
- [Commits](https://github.com/chalk/chalk/compare/v2.3.2...v2.4.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
build: bump webpack from 4.5.0 to 4.6.0

Bumps [webpack](https://github.com/webpack/webpack) from 4.5.0 to 4.6.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v4.5.0...v4.6.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

ncoden committed:
docs: replace "Yetinauts" mentions by "Core Team"

ncoden committed:
docs: improve issues & PR recommendation in CONTRIBUTE.md

ncoden committed:
docs: fix lists formatting in CONTRIBUTING.md

ncoden committed:
docs: fix link to CONTRIBUTING.md in README

ncoden committed:
docs: fix internals links in CONTRIBUTING.md

ncoden committed:
docs: fix typo in Contributions section of CONTRIBUTING.md

dependabot-support committed:
build: bump mocha from 5.1.0 to 5.1.1

Bumps [mocha](https://github.com/mochajs/mocha) from 5.1.0 to 5.1.1.
- [Release notes](https://github.com/mochajs/mocha/releases)
- [Changelog](https://github.com/mochajs/mocha/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mochajs/mocha/compare/v5.1.0...v5.1.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

Commit:
fix: typo in Reveal function.

dependabot-support committed:
build: bump browser-sync from 2.23.6 to 2.23.7

Bumps [browser-sync](https://github.com/BrowserSync/browser-sync) from 2.23.6 to 2.23.7.
- [Release notes](https://github.com/BrowserSync/browser-sync/releases)
- [Changelog](https://github.com/BrowserSync/browser-sync/blob/master/CHANGELOG.md)
- [Commits](https://github.com/BrowserSync/browser-sync/compare/v2.23.6...v2.23.7)

Signed-off-by: dependabot[bot] <support@dependabot.com>

rmx90210 committed:
Update xy-grid.md

Minor style edit

ncoden committed:
chore: improve GitHub issue template

This template was generated by https://www.talater.com/open-source-templates/#/ and modified according to our own CONTRIBUTION.md guidelines.

ncoden committed:
chore: improve GitHub pull request template

This template was generated by https://www.talater.com/open-source-templates/#/ and modified according to our own CONTRIBUTION.md guidelines.

ncoden committed:
docs: improve gitflow documentation in CONTRIBUTING.md

ncoden committed:
chore: improve GitHub templates readibility in text format

Changes:
* limite lines lenght to 80 characters
* add centering and left/right alignement of texts
* use `-` instead of `*` for lists

ncoden committed:
chore: simplify the description explainaition in GitHub pull request template

ncoden committed:
chore: add checkbox aobut commits in GitHub pull request template

ncoden committed:
chore: make the user "follow" CONTRIBUTING.md and not simply read it

ncoden committed:
docs: fix various typo in contribution recommendations

ncoden committed:
docs: improve README logo responsivness

ncoden committed:
chore: uncheck TODO lists in CONTRIBUTING.md

As there are things to do and not done, they should not be pre-checked.

ncoden committed:
docs: fix typo in CONTRIBUTING.md

ncoden committed:
docs: fix README logo inline styles for GitHub

GitHub remove all inline styles but `max-width: 100%`

ncoden committed:
chore: update yarn lockfile

ncoden reviewed (approved):
👍 LGTM, thank you @DanielRuf

ncoden commented:
I guess that tests fail here because of the focus bug you described at #11177 ?

DanielRuf commented:
> I guess that tests fail here because of the focus bug you described at #11177 ?

Exactly.

ncoden commented:
Ok. I'll take a look at this bug later. Like, why does this happens only now with updated test dependencies ?

DanielRuf commented:
> Ok. I'll take a look at this bug later. Like, why does this happens only now with updated test dependencies ?

It also happened before but the new dependencies clearly show the issue instead of hiding it. Before this we saw it only with the previous Browserstack setup and now the testsuite fails due to this (before this it got a timeout with an undefined exit code).

dependabot-support committed:
build: bump mocha-headless-chrome from 1.8.3 to 2.0.0

Bumps [mocha-headless-chrome](https://github.com/direct-adv-interfaces/mocha-headless-chrome) from 1.8.3 to 2.0.0.
- [Release notes](https://github.com/direct-adv-interfaces/mocha-headless-chrome/releases)
- [Commits](https://github.com/direct-adv-interfaces/mocha-headless-chrome/compare/v1.8.3...v2.0.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
build: bump chalk from 2.4.0 to 2.4.1

Bumps [chalk](https://github.com/chalk/chalk) from 2.4.0 to 2.4.1.
- [Release notes](https://github.com/chalk/chalk/releases)
- [Commits](https://github.com/chalk/chalk/compare/v2.4.0...v2.4.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>
